### PR TITLE
[modernize] readability-redundant-smartptr-get

### PR DESCRIPTION
**Pull request by Rechi**
## Description
run clang-tidy readability-redundant-smartptr-get check and applied the suggested fixes

## Types of change
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/cdrip/CDDARipper.cpp [hunk 1]
```diff
@@ -99,7 +99,7 @@ bool CCDDARipper::RipCD()
     CFileItemPtr pItem = vecItems[i];
     CMusicInfoTagLoaderFactory factory;
     std::unique_ptr<IMusicInfoTagLoader> pLoader (factory.CreateLoader(*pItem));
-    if (NULL != pLoader.get())
+    if (NULL != pLoader)
```
Comment (notspiff) [inline xbmc/cdrip/CDDARipper.cpp]: nullptr or drop comparison

File: xbmc/dbwrappers/Database.cpp [hunk 2]
```diff
@@ -247,7 +247,7 @@ std::string CDatabase::PrepareSQL(std::string strStmt, ...) const
 {
   std::string strResult = "";
 
-  if (NULL != m_pDB.get())
+  if (NULL != m_pDB)
```
Comment (notspiff) [inline xbmc/dbwrappers/Database.cpp]: nullptr or not comparison

File: xbmc/dbwrappers/Database.cpp [hunk 3]
```diff
@@ -336,8 +336,10 @@ bool CDatabase::ExecuteQuery(const std::string &strQuery)
 
   try
   {
-    if (NULL == m_pDB.get()) return bReturn;
-    if (NULL == m_pDS.get()) return bReturn;
+    if (NULL == m_pDB)
```
Comment (notspiff) [inline xbmc/dbwrappers/Database.cpp]: nullptr or no comparison

File: xbmc/dbwrappers/Database.cpp [hunk 4]
```diff
@@ -336,8 +336,10 @@ bool CDatabase::ExecuteQuery(const std::string &strQuery)
...
+      return bReturn;
+    if (NULL == m_pDS)
```
Comment (notspiff) [inline xbmc/dbwrappers/Database.cpp]: nullptr or no comparison

File: xbmc/dbwrappers/Database.cpp [hunk 5]
```diff
@@ -356,8 +358,10 @@ bool CDatabase::ResultQuery(const std::string &strQuery)
 
   try
   {
-    if (NULL == m_pDB.get()) return bReturn;
-    if (NULL == m_pDS.get()) return bReturn;
+    if (NULL == m_pDB)
```
Comment (notspiff) [inline xbmc/dbwrappers/Database.cpp]: nullptr or no comparison

File: xbmc/dbwrappers/Database.cpp [hunk 6]
```diff
@@ -356,8 +358,10 @@ bool CDatabase::ResultQuery(const std::string &strQuery)
...
+      return bReturn;
+    if (NULL == m_pDS)
```
Comment (notspiff) [inline xbmc/dbwrappers/Database.cpp]: nullptr or no comparison

File: xbmc/dbwrappers/Database.cpp [hunk 7]
```diff
@@ -379,8 +383,10 @@ bool CDatabase::QueueInsertQuery(const std::string &strQuery)
 
   if (!m_bMultiWrite)
   {
-    if (NULL == m_pDB.get()) return false;
-    if (NULL == m_pDS2.get()) return false;
+    if (NULL == m_pDB)
```
Comment (notspiff) [inline xbmc/dbwrappers/Database.cpp]: nullptr or no comparison

File: xbmc/dbwrappers/Database.cpp [hunk 8]
```diff
@@ -379,8 +383,10 @@ bool CDatabase::QueueInsertQuery(const std::string &strQuery)
...
+      return false;
+    if (NULL == m_pDS2)
```
Comment (notspiff) [inline xbmc/dbwrappers/Database.cpp]: nullptr or no comparison

File: xbmc/dbwrappers/Database.cpp [hunk 9]
```diff
@@ -596,8 +602,10 @@ void CDatabase::Close()
   m_openCount = 0;
   m_multipleExecute = false;
 
-  if (NULL == m_pDB.get() ) return ;
-  if (NULL != m_pDS.get()) m_pDS->close();
+  if (NULL == m_pDB)
```
Comment (notspiff) [inline xbmc/dbwrappers/Database.cpp]: nullptr or no comparison

File: xbmc/dbwrappers/Database.cpp [hunk 10]
```diff
@@ -596,8 +602,10 @@ void CDatabase::Close()
...
+    return;
+  if (NULL != m_pDS)
```
Comment (notspiff) [inline xbmc/dbwrappers/Database.cpp]: nullptr or no comparison

File: xbmc/dbwrappers/Database.cpp [hunk 11]
```diff
@@ -611,8 +619,10 @@ bool CDatabase::Compress(bool bForce /* =true */)
 
   try
   {
-    if (NULL == m_pDB.get()) return false;
-    if (NULL == m_pDS.get()) return false;
+    if (NULL == m_pDB)
```
Comment (notspiff) [inline xbmc/dbwrappers/Database.cpp]: nullptr or no comparison

File: xbmc/dbwrappers/Database.cpp [hunk 12]
```diff
@@ -611,8 +619,10 @@ bool CDatabase::Compress(bool bForce /* =true */)
...
+      return false;
+    if (NULL == m_pDS)
```
Comment (notspiff) [inline xbmc/dbwrappers/Database.cpp]: nullptr or no comparison

File: xbmc/dbwrappers/Database.cpp [hunk 13]
```diff
@@ -649,7 +659,7 @@ void CDatabase::BeginTransaction()
 {
   try
   {
-    if (NULL != m_pDB.get())
+    if (NULL != m_pDB)
```
Comment (notspiff) [inline xbmc/dbwrappers/Database.cpp]: nullptr or no comparison

File: xbmc/dbwrappers/Database.cpp [hunk 14]
```diff
@@ -662,7 +672,7 @@ bool CDatabase::CommitTransaction()
...
```
Comment (notspiff) [inline xbmc/dbwrappers/Database.cpp]: nullptr or no comparison

File: xbmc/dbwrappers/Database.cpp [hunk 15]
```diff
@@ -677,7 +687,7 @@ void CDatabase::RollbackTransaction()
...
```
Comment (notspiff) [inline xbmc/dbwrappers/Database.cpp]: nullptr or no comparison

File: xbmc/filesystem/FileCache.cpp [hunk 16]
```diff
@@ -212,7 +212,7 @@ void CFileCache::Process()
 
   // create our read buffer
   std::unique_ptr<char[]> buffer(new char[m_chunkSize]);
-  if (buffer.get() == NULL)
+  if (buffer == NULL)
```
Comment (notspiff) [inline xbmc/filesystem/FileCache.cpp]: nullptr or no comparison

File: xbmc/filesystem/PlaylistFileDirectory.cpp [hunk 17]
```diff
@@ -25,7 +25,7 @@ namespace XFILE
   {
     const std::string pathToUrl = url.Get();
     std::unique_ptr<CPlayList> pPlayList (CPlayListFactory::Create(pathToUrl));
-    if ( NULL != pPlayList.get())
+    if (NULL != pPlayList)
```
Comment (notspiff) [inline xbmc/filesystem/PlaylistFileDirectory.cpp]: nullptr or no comparison

File: xbmc/filesystem/PlaylistFileDirectory.cpp [hunk 18]
```diff
@@ -47,7 +47,7 @@ namespace XFILE
...
```
Comment (notspiff) [inline xbmc/filesystem/PlaylistFileDirectory.cpp]: nullptr or no comparison

File: xbmc/interfaces/legacy/PlayList.cpp [hunk 19]
```diff
@@ -70,7 +70,7 @@ namespace XBMCAddon
         // load a playlist like .m3u, .pls
         // first get correct factory to load playlist
         std::unique_ptr<CPlayList> pPlayList (CPlayListFactory::Create(item));
-        if ( NULL != pPlayList.get())
+        if (NULL != pPlayList)
```
Comment (notspiff) [inline xbmc/interfaces/legacy/PlayList.cpp]: nullptr or no comparison

File: xbmc/music/MusicInfoLoader.cpp [hunk 20]
```diff
@@ -126,7 +126,7 @@ bool CMusicInfoLoader::LoadAdditionalTagInfo(CFileItem* pItem)
   // fetch the lyrics and add it to the current music info tag
   CFileItem tempItem(path, false);
   std::unique_ptr<IMusicInfoTagLoader> pLoader (CMusicInfoTagLoaderFactory::CreateLoader(tempItem));
-  if (NULL != pLoader.get())
+  if (NULL != pLoader)
```
Comment (notspiff) [inline xbmc/music/MusicInfoLoader.cpp]: nullptr or no comparison

File: xbmc/music/MusicDatabase.cpp [hunk 21]
```diff
@@ -11119,8 +11286,10 @@ void CMusicDatabase::UpdateFileDateAdded(int songId, const std::string& strFileN
   CDateTime dateAdded;
   try
   {
-    if (NULL == m_pDB.get()) return;
-    if (NULL == m_pDS.get()) return;
+    if (NULL == m_pDB)
+      return;
+    if (NULL == m_pDS)
```
Comment (notspiff) [inline xbmc/music/MusicDatabase.cpp]: nullptr or no comparison

File: xbmc/music/MusicThumbLoader.cpp [hunk 22]
```diff
@@ -331,7 +331,7 @@ bool CMusicThumbLoader::GetEmbeddedThumb(const std::string &path, EmbeddedArt &a
   CFileItem item(path, false);
   std::unique_ptr<IMusicInfoTagLoader> pLoader (CMusicInfoTagLoaderFactory::CreateLoader(item));
   CMusicInfoTag tag;
-  if (NULL != pLoader.get())
+  if (NULL != pLoader)
```
Comment (notspiff) [inline xbmc/music/MusicThumbLoader.cpp]: nullptr or no comparison

File: xbmc/music/infoscanner/MusicInfoScanner.cpp [hunk 23]
```diff
@@ -574,7 +574,7 @@ CInfoScanner::INFO_RET CMusicInfoScanner::ScanTags(const CFileItemList& items,
     if (!tag.Loaded())
     {
       std::unique_ptr<IMusicInfoTagLoader> pLoader (CMusicInfoTagLoaderFactory::CreateLoader(*pItem));
-      if (NULL != pLoader.get())
+      if (NULL != pLoader)
```
Comment (notspiff) [inline xbmc/music/infoscanner/MusicInfoScanner.cpp]: nullptr or no comparison

File: xbmc/music/windows/GUIWindowMusicNav.cpp [hunk 24]
```diff
@@ -839,7 +840,7 @@ bool CGUIWindowMusicNav::GetSongsFromPlayList(const std::string& strPlayList, CF
   CLog::Log(LOGDEBUG,"CGUIWindowMusicNav, opening playlist [%s]", strPlayList.c_str());
 
   std::unique_ptr<CPlayList> pPlayList (CPlayListFactory::Create(strPlayList));
-  if ( NULL != pPlayList.get())
+  if (NULL != pPlayList)
```
Comment (notspiff) [inline xbmc/music/windows/GUIWindowMusicNav.cpp]: nullptr or no comparison

File: xbmc/pictures/GUIWindowPictures.cpp [hunk 25]
```diff
@@ -533,7 +533,7 @@ void CGUIWindowPictures::LoadPlayList(const std::string& strPlayList)
 {
   CLog::Log(LOGDEBUG,"CGUIWindowPictures::LoadPlayList()... converting playlist into slideshow: %s", strPlayList.c_str());
   std::unique_ptr<CPlayList> pPlayList (CPlayListFactory::Create(strPlayList));
-  if ( NULL != pPlayList.get())
+  if (NULL != pPlayList)
```
Comment (notspiff) [inline xbmc/pictures/GUIWindowPictures.cpp]: nullptr or no comparison

File: xbmc/playlists/PlayListPLS.cpp [hunk 26]
```diff
@@ -373,7 +373,7 @@ bool CPlayListASX::LoadData(std::istream& stream)
         if (!value.empty())
         { // found an entryref, let's try loading that url
           std::unique_ptr<CPlayList> playlist(CPlayListFactory::Create(value));
-          if (NULL != playlist.get())
+          if (NULL != playlist)
```
Comment (notspiff) [inline xbmc/playlists/PlayListPLS.cpp]: nullptr or no comparison

File: xbmc/video/VideoDatabase.cpp [hunk 27]
```diff
@@ -542,8 +542,10 @@ int CVideoDatabase::GetPathId(const std::string& strPath)
   try
   {
     int idPath=-1;
-    if (NULL == m_pDB.get()) return -1;
-    if (NULL == m_pDS.get()) return -1;
+    if (NULL == m_pDB)
```
Comment (notspiff) [inline xbmc/video/VideoDatabase.cpp]: nullptr or no comparison

File: xbmc/view/ViewDatabase.cpp [hunk 28]
```diff
@@ -116,8 +116,10 @@ bool CViewDatabase::GetViewState(const std::string &path, int window, CViewState
 {
   try
   {
-    if (NULL == m_pDB.get()) return false;
-    if (NULL == m_pDS.get()) return false;
+    if (NULL == m_pDB)
```
Comment (notspiff) [inline xbmc/view/ViewDatabase.cpp]: nullptr or no comparison

File: xbmc/windows/GUIWindowFileManager.cpp [hunk 29]
```diff
@@ -635,7 +635,7 @@ void CGUIWindowFileManager::OnStart(CFileItem *pItem, const std::string &player)
   {
     std::string strPlayList = pItem->GetPath();
     std::unique_ptr<CPlayList> pPlayList (CPlayListFactory::Create(strPlayList));
-    if (NULL != pPlayList.get())
+    if (NULL != pPlayList)
```
Comment (notspiff) [inline xbmc/windows/GUIWindowFileManager.cpp]: nullptr or no comparison

**Review by notspiff — COMMENTED** [review]
I gave up commenting on all of them, but there should be one comment per affected file.

**Comment by Rechi** [comment]
run clang-tidy-diff with modernize-use-nullptr check

**Pull request merged**